lib/Db/ChattyLLM/MessageMapper.php

```diff
@@ -101,6 +101,24 @@ public function getMessageById(int $messageId): Message {
 		return $this->findEntity($qb);
 	}
 
+	/**
+	 * @param int $sessionId
+	 * @param int $taskId
+	 * @return Message
+	 * @throws DoesNotExistException
+	 * @throws MultipleObjectsReturnedException
+	 * @throws \OCP\DB\Exception
+	 */
+	public function getMessageByTaskId(int $sessionId, int $taskId): Message {
+		$qb = $this->db->getQueryBuilder();
+		$qb->select(Message::$columns)
+			->from($this->getTableName())
+			->where($qb->expr()->eq('session_id', $qb->createPositionalParameter($sessionId, IQueryBuilder::PARAM_INT)))
+			->andWhere($qb->expr()->eq('task_id', $qb->createPositionalParameter($taskId, IQueryBuilder::PARAM_INT)));
+
+		return $this->findEntity($qb);
```

Comment:
only task_id should suffice here and the index should help
maybe it can be argued that regeneration would not be called that often so it is also fine to skip it.

Reply:
If we do this and don't add extra checks in `ChattyLLMController` we allow any user to get any message as long as they provide a valid session ID. They can ask for any task id.

So either we keep it like i did, or we do as you say but check that the message object we get has the correct session ID.

Reply:
yup, you're right, let's keep sessionId